src/core/qgssnappingconfig.h

```diff
@@ -172,6 +172,10 @@ class CORE_EXPORT QgsSnappingConfig
     //! set if the snapping on intersection is enabled
     void setIntersectionSnapping( bool enabled );
 
+    bool editVerticesOnAllLayers() const;
+
+    void setEditVerticesOnAllLayers( bool enabled );
+
```

Comment:
This setting should not be a part of QgsSnappingConfig as it is not related to snapping at all... It is an option specific to one particular tool...